src/nodes.ts

```diff
@@ -0,0 +1,31 @@
+import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+
+export interface Nodes {
+    all(): Iterable<JsonLdObj>;
+
+    add(node: JsonLdObj): Promise<void>;
+
+    get(id: Iri): Promise<JsonLdObj>;
+
+    has(id: Iri): Promise<boolean>;
+}
```

Comment:
Suggest we put this in its own module, and try something like:

```typescript
export default interface Articles extends AsyncIterable<JsonLdObj> {
  count(): Promise<number>;

  get(id: Iri): Promise<JsonLdObj>;

  set(article: JsonLdObj): Promise<void>;

  delete(id: Iri): Promise<void>;

  has(id: Iri): Promise<boolean>;
};
```

Reply:
`Articles` now extends `Iterable` we can implement `AsyncIterable` as needed later.